crates/cxx-qt-gen/src/generator/rust/property/setter.rs

```diff
@@ -19,46 +19,63 @@ pub fn generate(
-    let qualified_impl = type_names.rust_qualified(cpp_class_name_rust)?;
 
-    // Determine if unsafe is required due to an unsafe type
-    let has_unsafe = if syn_type_is_cxx_bridge_unsafe(cxx_ty) {
-        quote! { unsafe }
-    } else {
-        quote! {}
-    };
+    if let (Some(setter), Some(setter_wrapper)) = (&idents.setter, &idents.setter_wrapper) {
+        let setter_wrapper_cpp = setter_wrapper.cxx_unqualified();
+
+        let setter_rust = setter.rust_unqualified();
+        let ident = &idents.name.rust_unqualified();
+        let ident_str = ident.to_string();
 
-    Ok(RustFragmentPair {
-        cxx_bridge: vec![quote! {
-            extern "Rust" {
-                #[cxx_name = #setter_wrapper_cpp]
-                // TODO: Add #[namespace] of the QObject to the declaration
-                #has_unsafe fn #setter_rust(self: Pin<&mut #cpp_class_name_rust>, value: #cxx_ty);
+        // Generate a notify name if it was provided, otherwise return empty
+        let notify_binding = match &idents.notify {
+            Some(notify) => {
+                let notify_ident = notify.rust_unqualified();
+                quote! {self.as_mut().#notify_ident();}
             }
-        }],
-        implementation: vec![quote! {
-            impl #qualified_impl {
-                #[doc = "Setter for the Q_PROPERTY "]
-                #[doc = #ident_str]
-                pub fn #setter_rust(mut self: core::pin::Pin<&mut Self>, value: #qualified_ty) {
-                    use cxx_qt::CxxQtType;
-                    if self.#ident == value {
-                        // don't want to set the value again and reemit the signal,
-                        // as this can cause binding loops
-                        return;
+            None => quote! {},
+        };
+
+        let qualified_ty = syn_type_cxx_bridge_to_qualified(cxx_ty, type_names)?;
+        let qualified_impl = type_names.rust_qualified(cpp_class_name_rust)?;
+
+        // Determine if unsafe is required due to an unsafe type
+        let has_unsafe = if syn_type_is_cxx_bridge_unsafe(cxx_ty) {
+            quote! { unsafe }
+        } else {
+            quote! {}
+        };
+
+        Ok(Some(RustFragmentPair {
+            cxx_bridge: vec![quote! {
+                extern "Rust" {
+                    #[cxx_name = #setter_wrapper_cpp]
+                    // TODO: Add #[namespace] of the QObject to the declaration
+                    #has_unsafe fn #setter_rust(self: Pin<&mut #cpp_class_name_rust>, value: #cxx_ty);
+                }
+            }],
+            implementation: vec![quote! {
+                impl #qualified_impl {
+                    #[doc = "Setter for the Q_PROPERTY "]
+                    #[doc = #ident_str]
+                    pub fn #setter_rust(mut self: core::pin::Pin<&mut Self>, value: #qualified_ty) {
+                        use cxx_qt::CxxQtType;
+                        //TODO: could be optimised to remove this condition?
```

Comment:
right, might be worth doing not sure, lets leave a github comment here to discuss...

Reply:
left an issue to remember this: https://github.com/KDAB/cxx-qt/issues/1000

Reply:
Cool :-)